std/range/package.d

```diff
@@ -1681,25 +1681,25 @@ if (Ranges.length >= 2
     assert(c[0] == 42);
 }
 
-@safe nothrow pure @nogc unittest
+@safe nothrow pure /*@nogc*/ unittest
 {
     static struct RefAccessRange
     {
         int[] r;
-        ref front() @property { return r[0]; }
-        ref back() @property { return r[$ - 1]; }
+        scope ref front() @property { return r[0]; }
+        scope ref back() @property { return r[$ - 1]; }
         void popFront() { r = r[1 .. $]; }
         void popBack() { r = r[0 .. $ - 1]; }
         auto empty() @property { return r.empty; }
-        ref opIndex(size_t i) { return r[i]; }
+        scope ref opIndex(size_t i) { return r[i]; }
         auto length() @property { return r.length; }
         alias opDollar = length;
         auto save() { return this; }
     }
     static assert(isRandomAccessRange!RefAccessRange);
     static assert(isRandomAccessRange!RefAccessRange);
-    int[4] a = [4, 3, 2, 1];
-    int[2] b = [6, 5];
+    int[] a = [4, 3, 2, 1];
+    int[] b = [6, 5];
     auto c = chooseAmong(0, RefAccessRange(a[]), RefAccessRange(b[]));
```

Comment:
1) `[]` is done here to create a slice of the static array (it's not needed for dynamic arrays, they are already slices)
2) It feels like things like this here should be possible with DIP1000 :/

However, I looked at this problem myself two days ago and I couldn't find a better solution either.
So I think `version(DIP1000) {} else` at the top of the unittest might be better, because then we know that this still needs a look at, but aren't blocked by it.
What do you or others think?

Reply:
1. I tried both, a and a[] etc. and they both work; don't know why I added superfluous [] again, probably to be minimally invasive, expecting a change back to static arrays

Reply:
Yes, in order not to lose the @nogc and static array testing, "version(DIP1000) {} else" will be better, but I'll wait for a go

Reply:
Maybe https://github.com/D-Programming-Deimos/openssl needs an update to openssl 1.0.2g? That's what I have on my Kubuntu system since the day before yesterday

Reply:
How is this related? (and most likely - I think the Deimos packages aren't super-actively maintained)

Reply:
It relates to my reasoning why continuous-integration/jenkins/pr-merge might have failed:
...
Fetching openssl 1.1.6+1.0.1g (getting selected version)...
Package diet-ng can be upgraded from 1.4.3 to 1.4.5.
Package taggedalgebraic can be upgraded from 0.10.8 to 0.10.11.
Package stdx-allocator can be upgraded from 2.77.0 to 2.77.1.
Package vibe-d can be upgraded from 0.8.3-alpha.3 to 0.8.3.
Package memutils can be upgraded from 0.4.9 to 0.4.10.
Package vibe-core can be upgraded from 1.4.0-alpha.1 to 1.4.0.
Package eventcore can be upgraded from 0.8.27 to 0.8.34.
Use "dub upgrade" to perform those changes.
Non-optional dependency openssl of botan not found in dependency tree!?.
script returned exit code 2

Reply:
Oh that's an old Dub bug, see e.g.

https://github.com/dlang/dub/issues/1180
https://github.com/dlang/dub/issues/1112
https://github.com/dlang/dub/issues/986

Unfortunately Jenkins with all the projects it tests is quite "vulnerable" to random failures.
OTOH it has caught tons of regressions, so nothing is for free :/

Anyhow I restarted the job.
BTW feel free to ping me on Slack or the people in #phobos when you run into this again.